apps/src/code-studio/components/progress/ProgressDot.jsx

```diff
@@ -283,6 +279,8 @@ export const ProgressDot = Radium(React.createClass({
     // Account for both the level based concept of locked, and the progress based concept.
     const isLocked = status === LevelStatus.locked;
 
+    const tooltipId = _.uniqueId();
+
```

Comment:
Nice find on the tooltip error!

It seems like this approach could theoretically lead to worse performance, since react will never see the same ID even when the same tooltip is being shown a second time. If the problem is that`props.level.title` is sometimes absent, then perhaps you could use a unique ID only in when it is absent. However if the problem is non-unique values of `props.level.title` then that probably won't work.

I think it's also fine to do nothing, and just keep this in mind if we start observing bad performance moving the mouse in an area of the screen that causes many tooltips to show.